lib/perso.ml

```diff
@@ -4189,24 +4178,16 @@ let eval_transl conf base env upp s c =
   | _ -> Templ.eval_transl conf upp s c
 
 let level_in_list level lev_list =
-  let level = if List.hd lev_list >= 0 then level else -level in
-  if List.mem level lev_list then Some level else None
+  match lev_list with
+  | [] -> None
+  | _ ->
+      let level = if List.nth lev_list 0 >= 0 then level else -level in
```

Comment:
here we can avoid to use `List.nth`, and also we can use `abs` (absolute value):
```ocaml
  match lev_list with
  | [] -> None
  | level :: _l ->
      let level = abs level in
```

Reply:
J'avais noté, mais cela me semble plus compliqué! `lev_list` est une liste de nombres positifs pour les ascendants, et de nombres négatifs pour les descendants. `level` lui est toujours positif!!
Il faut cependant que je vérifie que la liste lev_list commence bien systématiquement à +1 (ascendants) et -1 (descendants) et non à 0 (self)

Reply:
donc il faudrait quelque chose comme:
```ocaml
if level = 0 then !GWPARAM.syslog `LOG_ERR "level is 0 but should be +/- 1" ;
let level = abs level in
...
```
?

Reply:
non, level est toujours positif (ou nul pour self)
C'est lev_list qui peut être une liste de nombres positifs ou négatifs!!
En fait, la bonne réponse, c'est 
```
  match lev_list with
  | [] -> None
  | _ -> List.find_opt (fun lvl -> abs lvl = level) lev_list

```
Ceci-dit, logger une erreur si `List.nth lev_list 0 = 0` est utile.
Je factorise ces deux fonctions "in" et "less_or_equal"
Peut on écrire cela de façon plus simple?

Reply:
ah ok je n'avais pas bien compris le code

je comprends pas trop l'histoire des +/- 1
mais je pense que l'on veut :
```ocaml
let level_in_list level lev_list =
  match lev_list with
  | [] -> None
  | fist_element :: _l -> 
if fist_element = 0 then ** log d'erreur ** else 
 List.find_opt (fun lvl -> abs lvl = level) lev_list
```